lib/asm-docs/generated/asm-docs-riscv64.ts

```diff
@@ -0,0 +1,6472 @@
+                "url": "https://five-embeddev.github.io/riscv-docs-html//riscv-user-isa-manual/Priv-v1.12/f.html#sec:single-float-compute"
+            };
+
+        case "FMIN.D":
+            return {
+                "html": "<div><span class=\"opcode\"><b>FMIN.D</b> rd, rs1, rs2</span><br><div><b>ISA</b>: d</div></div>",
+                "tooltip": "\n\n(ISA: d)",
+                "url": "https://five-embeddev.github.io/riscv-docs-html/"
+            };
+
+        case "FMIN.H":
+            return {
+                "html": "<div><span class=\"opcode\"><b>FMIN.H</b> rd, rs1, rs2</span><br><div><b>ISA</b>: </div></div>",
+                "tooltip": "\n\n(ISA: )",
+                "url": "https://five-embeddev.github.io/riscv-docs-html/"
+            };
+
+        case "FMIN.Q":
+            return {
+                "html": "<div><span class=\"opcode\"><b>FMIN.Q</b> rd, rs1, rs2</span><br><div><b>ISA</b>: q</div></div>",
+                "tooltip": "\n\n(ISA: q)",
+                "url": "https://five-embeddev.github.io/riscv-docs-html/"
+            };
+
+        case "FMIN.S":
+            return {
+                "html": "<div><span class=\"opcode\"><b>FMIN.S</b> rd, rs1, rs2</span><br><div>Floating-point minimum-number and maximum-number instructions <b>FMIN.S</b> and FMAX.S write, respectively, the smaller or larger of rs1 and rs2 to rd. For the purposes of these instructions only, the value - 0.0 is considered to be less than the value + 0.0. If both inputs are NaNs, the result is the canonical NaN. If only one operand is a NaN, the result is the non-NaN operand. Signaling NaN inputs set the invalid operation exception flag, even when the result is not NaN.<br>Note that in version 2.2 of the F extension, the <b>FMIN.S</b> and FMAX.S instructions were amended to implement the proposed IEEE 754-201x minimumNumber and maximumNumber operations, rather than the IEEE 754-2008 minNum and maxNum operations. These operations differ in their handling of signaling NaNs.</div><br><div><b>ISA</b>: f</div></div>",
+                "tooltip": "Floating-point minimum-number and maximum-number instructions FMIN.S and FMAX.S write, respectively, the smaller or larger of rs1 and rs2 to rd. For the purposes of these instructions only, the value - 0.0 is considered to be less than the value + 0.0. If both inputs are NaNs, the result is the canonical NaN. If only one operand is a NaN, the result is the non-NaN operand. Signaling NaN inputs set the invalid operation exception flag, even when the result is not NaN.\nNote that in version 2.2 of the F extension, the
```

Comment:
The definition of fmsub.h in opcodes.yyaml is

```
  fmsub.h:
    opcode:
    - fmsub.h
    - rd
    - rs1
    - rs2
    - rs3
    - rm
    - 26..25=2
    - 6..2=0x11
    - 1..0=3
    opcode_group: zfh
    opcode_args: *22
    iss_code:
    - require_either_extension(EXT_ZFH, EXT_ZHINX);
    - require_fp;
    - softfloat_roundingMode = RM;
    - WRITE_FRD_H(f16_mulAdd(FRS1_H, FRS2_H, f16(FRS3_H.v ^ F16_SIGN)));
    - set_fp_exceptions;
```

As opposed to others, it doesn't have a main_desc, where the ISA value is taken from, but it does have an `opcode_group` (zfh), I wonder if it shouldn't be saying `ISA: zfh` ? (I have no idea what it actually means)

fmv.h.z also have the opcode_group zfh, but that one actually has a main_desc and the description, so I don't know what the best option would be..

Reply:
You are correct. It should say "(ISA: zfh)".  zfh means "Standard Extension for 16-bit Half-Precision Binary Floating-Point Instructions".  To generate opcodes.yaml, I'm using the most recent opcodes metadata files, but the html from an older ratified spec version, so it's probably missing `fmsub.h` in the older documentation file. `fmv.h` was found in the documentation so `main_desc` is present. 

I'm not sure if any compiler will generate code for the zfh extension.

Reply:
You are correct. It should say "(ISA: zfh)". `zfh` means "Standard Extension for 16-bit Half-Precision Binary Floating-Point Instructions". To generate `opcodes.yaml`, I'm using the most recent opcodes metadata files, but the html from an older ratified spec version, so it's probably missing `fmsub.h` in the older documentation file. `fmv.h` was found in the documentation so `main_desc` is present. 

I'm not sure if any compiler will generate code for the `zfh` extension.